addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4759,72 +4759,51 @@ registry.SnippetSave = SnippetOptionWidget.extend({
      */
     saveSnippet: function (previewMode, widgetValue, params) {
         return new Promise(resolve => {
-            const dialog = new Dialog(this, {
-                title: _t("Save Your Block"),
-                size: 'small',
-                $content: $(qweb.render('web_editor.dialog.save_snippet', {
-                    currentSnippetName: _.str.sprintf(_t("Custom %s"), this.data.snippetName),
-                })),
-                buttons: [{
-                    text: _t("Save"),
-                    classes: 'btn-primary',
-                    close: true,
-                    click: async () => {
-                        const save = await new Promise(resolve => {
-                            Dialog.confirm(this, _t("To save a snippet, we need to save all your previous modifications and reload the page."), {
-                                buttons: [
-                                    {
-                                        text: _t("Save and Reload"),
-                                        classes: 'btn-primary',
-                                        close: true,
-                                        click: () => resolve(true),
-                                    }, {
-                                        text: _t("Cancel"),
-                                        close: true,
-                                        click: () => resolve(false),
-                                    }
-                                ]
+            const snippetKey = this.$target[0].dataset.snippet;
+            const defaultSnippetName = _.str.sprintf(_t("Custom %s"), this.data.snippetName);
+            let thumbnailURL;
+            this.trigger_up('snippet_thumbnail_url_request', {
+                key: snippetKey,
+                onSuccess: url => thumbnailURL = url,
+            });
```

Comment:
Same as the old code: you don't need to do all that if the user says no to the save & reload, so it should only be done when he says yes.